src/structure/metrics.py

```diff
@@ -2,13 +2,37 @@
 from __future__ import annotations
 from typing import Optional
 import numpy as np
+from numpy.core.multiarray import ascontiguousarray
 import pandas as pd
 import biotite.structure as struc
 from .structure_context import Context, register_metric
 from . import pdbtm
 from .utils import residue_key, is_heavy, get_metadata_cols
+from .utils import build_sites_biotite as _build_sites_biotite, detect_hbonds as _detect_hbonds
 
-def calculate_sasa(array: struc.AtomArray | Context, vdw_radii: str = "ProtOr") -> pd.DataFrame:
+# TODO: move helper functions to separate file so only @registered_metric functions remain here
+def calculate_secondary_structure(array: struc.AtomArray | Context) -> np.ndarray:
+    """
+    Calculate secondary structure assignment per residue.
+    
+    Parameters:
+    -----------
+    array : AtomArray or Context
+        Structure array (amino acids only recommended) or Context.
+    
+    Returns:
+    --------
+    np.ndarray
+        Per-residue secondary structure assignment.
+        'a' = alpha-helix, 'b' = beta-sheet, 'c' = coil
+    """
+    if isinstance(array, Context):
```

Comment:
I don't think we'll need to pass a context object, this helper gets called by the metric functions